lib/Yield/ControlFlow/SVG.cpp

```diff
@@ -98,81 +54,93 @@ static std::string_view edgeTypeAsString(yield::Graph::EdgeType Type) {
-    Points += llvm::formatv(templates::BezierCubic,
-                            First.X + XDistance * BendFactor,
-                            -First.Y,
-                            Second.X - XDistance * BendFactor,
-                            -Second.Y,
-                            Second.X,
-                            -Second.Y);
+    revng_assert(Path.size() >= 2);
+    for (auto Iter = Path.begin(); Iter != std::prev(Path.end()); ++Iter)
+      Points += cubicBend(*Iter, *std::next(Iter), UseVerticalCurves);
   }
 
   revng_assert(!Points.empty());
   revng_assert(Points.back() == ' ');
   Points.pop_back(); // Remove an extra space at the end.
-  Points += '"';
 
-  return llvm::formatv(templates::Edge, edgeTypeAsString(Type), Points);
-}
-
-static std::string content(const yield::Node *Node,
-                           const yield::Function &Function,
-                           const model::Binary &Binary) {
-  revng_assert(Node != nullptr);
-
-  if (Node->Address.isValid()) {
-    // A normal node.
-    return yield::ptml::controlFlowNode(Node->Address, Function, Binary);
-  } else {
-    // The entry/exit/error node.
-    return "";
-  }
+  std::string Marker = llvm::formatv("url(#{0}-arrow-head)",
+                                     edgeTypeAsString(Type));
+  return ::ptml::Tag("path")
+    .addAttribute("class", std::string(edgeTypeAsString(Type)) += "-edge")
+    .addAttribute("d", std::move(Points))
+    .addAttribute("marker-end", std::move(Marker))
+    .addAttribute("fill", "none")
+    .serialize();
 }
 
 static std::string node(const yield::Node *Node,
-                        const yield::Function &Function,
-                        const model::Binary &Binary,
+                        std::string &&Content,
                         const yield::cfg::Configuration &Configuration) {
   yield::Graph::Size HalfSize{ Node->Size.W / 2, Node->Size.H / 2 };
   yield::Graph::Point TopLeft{ Node->Center.X - HalfSize.W,
                                -Node->Center.Y - HalfSize.H };
 
-  auto Text = llvm::formatv(templates::ForeignObject,
-                            tags::NodeContents,
-                            TopLeft.X,
-                            TopLeft.Y,
-                            Node->Size.W,
-                            Node->Size.H,
-                            content(Node, Function, Binary));
-
-  auto Body = llvm::formatv(templates::Rect,
-                            tags::NodeBody,
-                            TopLeft.X,
-                            TopLeft.Y,
-                            Configuration.NodeCornerRoundingFactor,
-                            Node->Size.W,
-                            Node->Size.H);
-
-  return llvm::formatv("{0}", std::move(Text) + std::move(Body));
+  ::ptml::Tag Body("body", std::move(Content));
+  Body.addAttribute("xmlns", R"("http://www.w3.org/1999/xhtml")");
+
+  ::ptml::Tag Text("foreignObject", Body.serialize());
```

Comment:
You're using `Tag` a lot, put a `using`.

Reply:
Something like this? https://github.com/revng/revng/pull/191/commits/115333a2656cf549b82cb4e37cdce87e4b48c4ab

Reply:
Would just using `Tag` trigger any ambiguity? Let's avoid introducing new names if not necessary.
We're in SVG.cpp, I'm pretty sure people realize we're talking about *ptml* tags.

Reply:
Yeah, I guess. Mostly it's just hesitation about introducing a new three-letter keyword into a global namespace. Even if it's cpp-file-local.

I'll do it though. `Tag`'s a much better name after all.

Reply:
Done: https://github.com/revng/revng/pull/191/commits/3a027e8c965f5389548872368b724dc59cc39579